components/team_members_dropdown/team_members_dropdown.test.tsx

```diff
@@ -0,0 +1,224 @@
+        password: 'horsebatterystaple',
+        auth_data: '',
+        auth_service: '',
+        email: 'bonjour@monde.com',
+        email_verified: true,
+        nickname: 'username2',
+        first_name: 'Bonjour',
+        last_name: 'Monde',
+        position: '',
+        roles: 'team_admin',
+        allow_marketing: true,
+        props: {},
+        notify_props: {
+            desktop: 'all' as const,
+            desktop_sound: 'true' as const,
+            email: 'true' as const,
+            mark_unread: 'all' as const,
+            push: 'default' as const,
+            push_status: 'ooo' as const,
+            comments: 'root' as const,
+            first_name: 'true' as const,
+            channel: 'true' as const,
+            mention_keys: '' as const
+        },
+        last_password_update: 0,
+        last_picture_update: 0,
+        failed_attempts: 0,
+        locale: '',
+        mfa_active: true,
+        mfa_secret: '',
+        last_activity_at: 0,
+        is_bot: false,
+        bot_description: '',
+        bot_last_icon_update: 0,
+        terms_of_service_id: '',
+        terms_of_service_create_at: 0,
+    };
+
+    const bot = {
+        id: 'bot-user',
+        create_at: 0,
+        update_at: 0,
+        delete_at: 0,
+        username: 'bot',
+        password: 'horsebatterystaple',
+        auth_data: '',
+        auth_service: '',
+        email: 'bot@bot.com',
+        email_verified: true,
+        nickname: 'bot',
+        first_name: 'Bot',
+        last_name: '',
+        position: '',
+        roles: 'system_user',
+        allow_marketing: true,
+        props: {},
+        notify_props: {
+            desktop: 'all' as const,
+            desktop_sound: 'true' as const,
+            email: 'true' as const,
+            mark_unread: 'all' as const,
+            push: 'default' as const,
+            push_status: 'ooo' as const,
+            comments: 'root' as const,
+            first_name: 'true' as const,
+            channel: 'true' as const,
+            mention_keys: '' as const
+        },
+        last_password_update: 0,
+        last_picture_update: 0,
+        failed_attempts: 0,
+        locale: '',
+        mfa_active: true,
+        mfa_secret: '',
+        last_activity_at: 0,
+        is_bot: true,
+        bot_description: '',
+        bot_last_icon_update: 0,
+        terms_of_service_id: '',
+        terms_of_service_create_at: 0,
+    };
+
+    const team = {
+        id: 'qkh1bxwgkjry3x4sgn5roa8ypy',
+        create_at: 1553808980710,
+        update_at: 1553808980710,
+        delete_at: 0,
+        display_name: 'engineering',
+        name: 'engineering',
+        description: '',
+        email: 'sysadmin@test.com',
+        type: 'O' as const,
+        company_name: '',
+        allowed_domains: '',
+        invite_id: 'sgic8xqghb8iupttw6skeqifuo',
+        allow_open_invite: false,
+        scheme_id: 'serjktghwaljreglaw43q34qtl',
+        group_constrained: false,
```

Comment:
Lots of props added to `.test.tsx` to satisfy the types' constraints. `group_constrained` wasn't defined previously.  After studying the tests, this field was assumed not existent. Here I assumed this field is `false` initially.

Reply:
This is something I'm still unsure of with the TypeScript migration. Having unit tests in TypeScript means we have to create test objects more rigorously, but it does make the tests for missing props and such more useful.

Reply:
It's great that @devinbinnie pointed out that `utils/test_helper.ts` could be used for creating these test objects/mocks. 👍 

For the Team object, there's no optional props. The test object should include all props including `group_constrained`.